packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -40,7 +40,7 @@ class RenderProxyBox extends RenderBox with RenderObjectWithChildMixin<RenderBox
   /// Proxy render boxes are rarely created directly because they simply proxy
   /// the render box protocol to [child]. Instead, consider using one of the
   /// subclasses.
-  RenderProxyBox([RenderBox child = null]) {
+  RenderProxyBox([RenderBox child = null]) { //ignore: avoid_init_to_null
```

Comment:
Some notes:

I cannot reproduce this with the command line analyzer.  Running from bleeding edge, or from 1.25.0-dev.9.0, analyzing packages/flutter/lib/src/rendering/proxy_box.dart, I get no override errors with or without the `= null`.

Using Intellij, manually setting the dart sdk: 
  - If I point the dart sdk to 1.25-dev.9.0 I see the behavior described above.  I get a long list of overrider errors without `= null`, and some (but not all of them) go away if I add the `= null` back in.
 - If I point the dart sdk to bleeding edge, the variance caused by adding removing the unrelated code goes away, but I still see the errors.

I have extracted out the relevant class and override hierarchy into a single file example, but cannot reproduce with either the command line analyzer nor with Intellij (see code below).

The errors that I am seeing are consistent with a missing `covariant` on the base method, but the base method is marked `covariant`. 

If I remove the `covariant` from base method for applyPaintTransform, then the command line analyzer gives the same errors.

@bwilkerson I suspect that this is a bug specific to analysis server, and that specifically it may somehow be losing track of the `covariant` property.  I don't know what the code paths for the server are these days - this used to be computed during strong mode override inference in the task model.

My attempt to extract out a repo produced this code (which doesn't produce the error):

```dart
class Matrix4 {}

abstract class RenderObject {
void applyPaintTransform(covariant RenderObject child, Matrix4 transform) {
  }
}

abstract class RenderBox extends RenderObject {
  void applyPaintTransform(RenderObject child, Matrix4 transform) {}
}

abstract class RenderObjectWithChildMixin<ChildType extends RenderObject> extends RenderObject {
  ChildType _child;
  ChildType get child => _child;
  set child(ChildType value) {}
}

abstract class RenderProxyBoxMixin extends RenderBox with RenderObjectWithChildMixin<RenderBox> {
  @override
  void applyPaintTransform(RenderObject child, Matrix4 transform) { }

}

class RenderProxyBox extends RenderBox with RenderObjectWithChildMixin<RenderBox>, RenderProxyBoxMixin {
  /// Creates a proxy render box.
  ///
  /// Proxy render boxes are rarely created directly because they simply proxy
  /// the render box protocol to [child]. Instead, consider using one of the
  /// subclasses.
  RenderProxyBox([RenderBox child = null]) {
    this.child = child;
  }
}


class RenderTransform extends RenderProxyBox {
@override
  void applyPaintTransform(RenderBox child, Matrix4 transform) {
  }
}
```

leafp-macbookpro:flutter leafp$ ../../bin/flutter doctor
[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-US, channel master)
    • Flutter at /Users/leafp/src/flutter
    • Framework revision 00e135d879 (68 minutes ago), 2017-08-04 12:01:35 -0700
    • Engine revision 1de56a33d2
    • Tools Dart version 1.25.0-dev.7.0

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.1

[-] Android Studio
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Ultimate Edition (version 2017.1.3)
    • Flutter plugin version 16.0
    • Dart plugin version 171.4249.16

[✓] Connected devices
    • None

leafp-macbookpro:flutter leafp$